src/test/java/org/assertj/core/api/localdatetime/LocalDateTimeAssert_hasYear_Test.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2022 the original author or authors.
+ */
+package org.assertj.core.api.localdatetime;
+
+import static java.lang.String.format;
+import static org.assertj.core.api.AbstractLocalDateTimeAssert.NULL_LOCAL_DATE_TIME_PARAMETER_MESSAGE;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
+import static org.assertj.core.api.BDDAssertions.then;
+import static org.assertj.core.util.AssertionsUtil.expectAssertionError;
+import static org.assertj.core.util.FailureMessages.actualIsNull;
+
+import java.time.LocalDateTime;
+
+import org.junit.jupiter.api.Test;
+
+
+public class LocalDateTimeAssert_hasYear_Test {
+  private final LocalDateTime refLocalDateTime = LocalDateTime.now();
+
+  @Test
+  void should_pass_if_actual_is_in_year_2022() {
+    assertThat(refLocalDateTime).hasYear(2022);
```

Comment:
I don't think this test is going to succeed in 2023 as `refLocalDateTime` is set to `now()` ;-)
Instead initialize it with `LocalDateTime.of()` where you can set explicitly the year, month, ... values.

while we are at it, rename `refLocalDateTime` to `actual` or `actualLocalDateTime` (use wichever you like best)